src/SkeletalTrapezoidation.cpp

```diff
@@ -0,0 +1,3083 @@
+            edge->from = v0;
+            edge->to = v1;
+//             edge->twin = nullptr;
+            edge->from->some_edge = edge;
+            
+            if (prev_edge)
+            {
+                edge->prev = prev_edge;
+                prev_edge->next = edge;
+            }
+            
+            prev_edge = edge;
+            p0 = p1;
+            v0 = v1;
+            
+            if (p1_idx < discretized.size() - 1)
+            { // rib for last segment gets introduced outside this function!
+                bool is_next_to_start_or_end = false; // only ribs at the end of a cell should be skipped
+                make_rib(prev_edge, start_source_point, end_source_point, is_next_to_start_or_end);
+            }
+        }
+        assert(prev_edge);
+        vd_edge_to_he_edge.emplace(&vd_edge, prev_edge);
+    }
+}
+
+void SkeletalTrapezoidation::make_rib(edge_t*& prev_edge, Point start_source_point, Point end_source_point, bool is_next_to_start_or_end)
+{
+    Point p = LinearAlg2D::getClosestOnLineSegment(prev_edge->to->p, start_source_point, end_source_point);
+    coord_t dist = vSize(prev_edge->to->p - p);
+    prev_edge->to->data.distance_to_boundary = dist;
+    assert(dist >= 0);
+
+    graph.nodes.emplace_front(SkeletalTrapezoidationJoint(), p);
+    node_t* node = &graph.nodes.front();
+    node->data.distance_to_boundary = 0;
+    
+    graph.edges.emplace_front(SkeletalTrapezoidationEdge(SkeletalTrapezoidationEdge::EXTRA_VD));
+    edge_t* forth_edge = &graph.edges.front();
+    graph.edges.emplace_front(SkeletalTrapezoidationEdge(SkeletalTrapezoidationEdge::EXTRA_VD));
+    edge_t* back_edge = &graph.edges.front();
+    
+    prev_edge->next = forth_edge;
+    forth_edge->prev = prev_edge;
+    forth_edge->from = prev_edge->to;
+    forth_edge->to = node;
+    forth_edge->twin = back_edge;
+    back_edge->twin = forth_edge;
+    back_edge->from = node;
+    back_edge->to = prev_edge->to;
+    node->some_edge = back_edge;
+    
+    prev_edge = back_edge;
+}
+
+std::vector<Point> SkeletalTrapezoidation::discretize(const vd_t::edge_type& vd_edge, const std::vector<Point>& points, const std::vector<Segment>& segments)
+{
+    const vd_t::cell_type* left_cell = vd_edge.cell();
+    const vd_t::cell_type* right_cell = vd_edge.twin()->cell();
+    Point start = VoronoiUtils::p(vd_edge.vertex0());
+    Point end = VoronoiUtils::p(vd_edge.vertex1());
+    
+    bool point_left = left_cell->contains_point();
+    bool point_right = right_cell->contains_point();
+    if ((!point_left && !point_right)
+        || vd_edge.is_secondary() // source vert is directly connected to source segment
+    )
+    {
+        return std::vector<Point>({ start, end });
+    }
+    else if (point_left == !point_right)
+    {
+        const vd_t::cell_type* point_cell = left_cell;
+        const vd_t::cell_type* segment_cell = right_cell;
+        if (!point_left)
+        {
+            std::swap(point_cell, segment_cell);
```

Comment:
Why swap them? It's not that much effort to repeat the `getSourcePoint` and `getSourceSegment` for each side, right?